src/naive/date.rs

```diff
@@ -692,6 +680,83 @@ impl NaiveDate {
         NaiveDate::from_mdf(year, Mdf::new(month as u32, day, flags))
     }
 
+    /// Subtract a duration in [`Months`] from the date
+    ///
+    /// If the day would be out of range for the resulting month, use the last day for that month.
+    ///
+    /// Returns `None` if the resulting date would be out of range.
+    ///
+    /// ```
+    /// # use chrono::{NaiveDate, Months};
+    /// assert_eq!(
+    ///     NaiveDate::from_ymd(2022, 2, 20).checked_sub_months(Months::new(6)),
+    ///     Some(NaiveDate::from_ymd(2021, 8, 20))
+    /// );
+    /// ```
+    pub fn checked_sub_months(self, months: Months) -> Option<Self> {
+        if months.0 == 0 {
+            return Some(self);
+        }
+
+        // Copy `i32::MIN` here so we don't have to do a complicated cast
+        match months.0 <= 2_147_483_648 {
```

Comment:
There's a bug here, best make it `2_147_483_647` https://github.com/chronotope/chrono/pull/778#discussion_r947706844

Reply:
this code was accidentally moved, I've now rebased to include your recently merged fixes